### DISCUSSION TITLE
Schema for Coverage Analysis Outputs

### DISCUSSION THREAD

**Discussion by dselva80** [eose-tools-org/eose-api#8] [General]
- IntermediateCoverageAnalysisResponse: GeoJSON Feature Collection, List of observations for each sample point, List of revisit durations for each sample point
- FinalCoverageAnalysisResponse: Harmonic mean revisit time

**Comment by ptgrogan** [comment]
The [example TAT-C notebook ](https://github.com/code-lab-org/tatc/blob/main/docs/examples/ComputeCoverage.ipynb) shows typical coverage analysis. Code block 11 shows the format of final analysis results in the form of a GeoDataFrame which has a 1-1 mapping to GeoJSON FeatureCollection. The geometry describes the sample point. The features describe analysis results like revisit and number samples. The final analysis results would reduce the spatial data to a single value, like harmonic_mean_revisit.

**Comment by ptgrogan** [comment]
I added a prototype [CoverageResponse implementation](https://github.com/eose-tools-org/eose-api/blob/dev-tatc-analysis/src/eose/coverage.py) with an [example TAT-C implementation/demonstration](https://github.com/eose-tools-org/eose-api/blob/dev-tatc-analysis/src/tatc-coverage-example.ipynb).

This version reports both the top-level metrics over the entire domain as well as the results for each target point.

**Comment by ptgrogan** [comment]
Split out the aggregation (revisit/sample size) across constellation members and reduction (top-level harmonic mean revisit and coverage fraction) as separate analysis interfaces.
